### PR TITLE
Allow `Series.select/3` on_true and on_false both to receive series of size 1

### PR DESCRIPTION
**Pull request by sasikumar87**
Related issue https://github.com/elixir-nx/explorer/issues/546

### CODE REVIEW

**Review by philss — APPROVED** [review]
Oh, I see. Makes sense! :shipit:

File: lib/explorer/polars_backend/series.ex [hunk 1]
```diff
@@ -149,13 +149,15 @@ defmodule Explorer.PolarsBackend.Series do
     predicate_size = size(predicate)
     on_true_size = size(on_true)
     on_false_size = size(on_false)
+    singleton_condition = on_true_size == 1 or on_false_size == 1
 
-    if on_true_size != on_false_size and on_true_size != 1 and on_false_size != 1 do
+    if on_true_size != on_false_size and !singleton_condition do
```
Comment (josevalim) [inline lib/explorer/polars_backend/series.ex:154]:
```suggestion
    if on_true_size != on_false_size and not singleton_condition do
```

File: lib/explorer/polars_backend/series.ex [hunk 2]
```diff
@@ -149,13 +149,15 @@ defmodule Explorer.PolarsBackend.Series do
...
       raise ArgumentError,
             "series in select/3 must have the same size or size of 1, got: #{on_true_size} and #{on_false_size}"
     end
 
-    if predicate_size != 1 and predicate_size != on_true_size and predicate_size != on_false_size do
+    if predicate_size != 1 and predicate_size != on_true_size and predicate_size != on_false_size and
+         !singleton_condition do
```
Comment (josevalim) [inline lib/explorer/polars_backend/series.ex:160]:
```suggestion
         not singleton_condition do
```

**Comment by josevalim** [comment]
Beautiful! :green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**